File: WebAssembly.Server/Models/MonthlyOverview.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

public class MonthlyOverview
{
    [Key]
    public string Id { get; set; } = Guid.NewGuid().ToString();

    public string GroupId { get; set; } = "";
    public string MonthKey { get; set; } = ""; // "2025-07"
    public string YearKey { get; set; } = "";  // "2025" ✅ NEU

    public string Name { get; set; } = "";
    public string Status { get; set; } = "open";

    public decimal Total { get; set; }
    public decimal Shared { get; set; }
    public decimal Child { get; set; }
    public decimal Balance { get; set; }
    
    public List<Expense>? Expenses { get; set; } // NEU

    [NotMapped] public Dictionary<string, decimal> SharedByUser { get; set; } = new();
    [NotMapped] public Dictionary<string, decimal> ChildByUser { get; set; } = new();
    [NotMapped] public Dictionary<string, decimal> TotalByUser { get; set; } = new();
    [NotMapped] public Dictionary<string, decimal> BalanceByUser { get; set; } = new();
    [NotMapped] public Dictionary<string, bool> RejectionsByUser { get; set; } = new();
    [NotMapped]
    public Dictionary<string, bool> ConfirmationsByUser { get; set; } = new();
}

File: frontendvite/src/types/monthly-overview.ts
// types/monthly-overview.ts

export enum ClarificationStatus {
    Accepted = 'Accepted', // Zustimmung zur Ausgabe
    Rejected = 'Rejected', // Ablehnung / Widerspruch
    // Optional: InDiscussion = 'InDiscussion', Ignored = 'Ignored'
}

export interface ClarificationReaction {
    id: string
    expenseId: string
    userId: string
    status: ClarificationStatus // jetzt ein String-Enum
    timestamp: string // ISO-Format z. B. "2025-05-28T12:00:00Z"
}

export interface MonthlyOverview {
    id: string // GUID – eindeutiger Monatsbezeichner
    groupId: string
    monthId: string
    monthKey: string // "2025-07"
    yearKey: string // "2025"
    status:
        | 'open'
        | 'completed'
        | 'future'
        | 'pending'
        | 'notTakenIntoAccount'
        | 'needs-clarification'
        | 'past'

    name?: string

    total: number
    shared: number
    child: number
    balance: number

    sharedByUser?: Record<string, number>
    childByUser?: Record<string, number>
    totalByUser?: Record<string, number>
    balanceByUser?: Record<string, number>
    rejectionsByUser?: Record<string, boolean>
}
export interface YearOverview {
    year: number
    months: MonthlyOverview[]
}
